lib/pure/nimprof.nim

```diff
@@ -26,19 +26,22 @@ const
   withThreads = compileOption("threads")
   tickCountCorrection = 50_000
 
-when not declared(system.StackTrace):
-  type StackTrace = object
-    lines: array[0..20, cstring]
-    files: array[0..20, cstring]
-  proc `[]`*(st: StackTrace, i: int): cstring = st.lines[i]
-
 # We use a simple hash table of bounded size to keep track of the stack traces:
 type
   ProfileEntry = object
     total: int
     st: StackTrace
   ProfileData = array[0..64*1024-1, ptr ProfileEntry]
 
+proc `==`(x, y: cstring): bool {.noSideEffect, inline.} =
+  ## Checks for equality between two `cstring` variables.
+  # copied over from system.nim as workaround for #10106
+  proc strcmp(a, b: cstring): cint {.noSideEffect,
+    importc, header: "<string.h>".}
+  if pointer(x) == pointer(y): result = true
+  elif x.isNil or y.isNil: result = false
+  else: result = strcmp(x, y) == 0
+
```

Comment:
is this still a letfover, I don't see you use it anywhere.

Reply:
that's the fix for #10106 ; without it it fails:
```
nim c --profiler:on compiler/nim.nim
compiler/nim c -o:/tmp/z01  compiler/nim.nim
```
fails with error msg here https://github.com/nim-lang/Nim/issues/10106#issue-394491093

Reply:
This is not a fix, it's workaround and I don't like it.